### ISSUE TITLE
Docs feedback: decision_cache is documented as a block but behaves as an argument

### ISSUE THREAD

**Issue by clayton-cornell** [grafana/alloy#1813]
### URL

https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.processor.tail_sampling/#otelcolprocessortail_sampling

### Feedback

Slack discussion: https://raintank-corp.slack.com/archives/CSN5HV0CQ/p1727891950877629

According to the documentation, `decision_cache` is a block, and would be added to the `alloy.config` file as:

```
otelcol.processor.tail_sampling "LABEL" {
	decision_cache {
		sampled_cache_size = 100000
		}
...
```

This configuration generates an error similar to:

```
327 | decision_cache {
| _____^^^^^^^^^^^^^^^^
328 | | sampled_cache_size = 100000
329 | | }
| |_________^
```

The configuration works as expected if you add `decision_cache` formatted as an argument like this:

```
	decision_cache = {
		sampled_cache_size = 100000,
		}
```

And confirmed here:

![Image](https://github.com/user-attachments/assets/be969d43-508e-46e5-a6eb-b175590c3ec3)

If this is the correct behaviour, the documentation needs to be updated.

**Comment by jdbaldry** [comment]
I'm not sure about the "argument" terminology, I think the syntax reference calls these "attributes": https://grafana.com/docs/alloy/latest/get-started/configuration-syntax/syntax/#attributes

**Comment by alexwulf** [comment]
Adding full error message received for future searchability:
```
Error: C:\Program Files\GrafanaLabs\Alloy\config.alloy:327:2: "decision_cache" must be an attribute, but is used as a block

327 | decision_cache {
| _____^^^^^^^^^^^^^^^^
328 | | sampled_cache_size = 100000
329 | | }
| |_________^
```

@jdbaldry is correct - I was saying `argument` in the slack thread by mistake, but was meaning to say `attribute`.

**Comment by mattdurham** [comment]
This should have been a block, but its encoded in the config now and we cant change it without breaking backwards compat.

**Comment by clayton-cornell** [comment]
@mattdurham I've opened a PR to update the docs to reflect how this was implemented in the code. Do we want to change the docs, keeping it as an attribute, or fix `decision_cache` in the code to promote it to a block.

**Comment by mattdurham** [comment]
IMO changing the code breaks the compatibility, which we could justify saying it’s a bug but I don’t feel good with that route.

**Comment by ptodev** [comment]
I agree that this should have been a block. I wonder if we can have both an attribute and a block called "decision_cache"? Then we can deprecate the argument.
